src/components/Modal/Modal.stories.tsx

```diff
@@ -0,0 +1,218 @@
+import React from 'react'
+
+import { Modal } from './Modal'
+import { ModalHeading } from './ModalHeading/ModalHeading'
+import { ModalFooter } from './ModalFooter/ModalFooter'
+import { ModalOpenButton } from './ModalOpenButton'
+
+import { Button } from '../Button/Button'
+import { ButtonGroup } from '../ButtonGroup/ButtonGroup'
+import { useModal } from './utils'
+
+export default {
+  title: 'Components/Modal',
+  component: Modal,
+  parameters: {
+    docs: {
+      description: {
+        component: `
+### USWDS 2.0 Modal component
+
+Source: http://designsystem.digital.gov/components/modal
+`,
+      },
+    },
+  },
+}
+
+export const defaultModal = (): React.ReactElement => {
+  const { isOpen, openModal, closeModal } = useModal()
+
+  return (
+    <>
+      <ModalOpenButton
+        handleOpen={openModal}
+        href="#example-modal-1"
+        aria-controls="example-modal-1"
+        onClick={(e) => e.preventDefault()}>
+        Open default modal
+      </ModalOpenButton>
+      <Modal
+        isOpen={isOpen}
+        closeModal={closeModal}
+        id="example-modal-1"
+        aria-labelledby="modal-1-heading"
+        aria-describedby="modal-1-description">
+        <ModalHeading id="modal-1-heading">
+          Are you sure you want to continue?
+        </ModalHeading>
+        <div className="usa-prose">
+          <p id="modal-1-description">
+            You have unsaved changes that will be lost.
+          </p>
+        </div>
```

Comment:
I'm assuming this wasn't encapsulated via `header` and `text` props (or similar) to make the `Modal` body more flexible for consumers?

Reply:
> I'm assuming this wasn't encapsulated via `header` and `text` props (or similar) to make the `Modal` body more flexible for consumers?

Yup correct, ModalHeading & ModalFooter just apply the markup in the USWDS examples and they are totally optional to use if someone needs to do something more different